201-cdn-with-storage-account/azuredeploy.json

```diff
@@ -1,76 +1,81 @@
+      },
       "properties": {
         "sku": {
           "name": "Standard"
         }
-      }
-    },
-    {
-      "apiVersion": "[variables('cdnApiVersion')]",
-      "dependsOn": [
-        "[concat('Microsoft.Cdn/profiles/', parameters('profileName'))]",
-        "[concat('Microsoft.Storage/storageAccounts/', parameters('storageAccountName'))]"
-      ],
-      "location": "[resourceGroup().location]",
-      "name": "[concat(parameters('profileName'),'/',parameters('endpointName'))]",
-      "type": "Microsoft.Cdn/profiles/endpoints",
-      "properties": {
-        "originHostHeader": "[replace(replace(reference(resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName')),'2015-06-15' ).primaryEndpoints.blob,'https://',''),'/','')]",
-        "isHttpAllowed": true,
-        "isHttpsAllowed": true,
-        "queryStringCachingBehavior": "IgnoreQueryString",
-        "contentTypesToCompress": [ "text/plain", "text/html", "text/css", "application/x-javascript", "text/javascript" ],
-        "isCompressionEnabled": "True",
-        "origins": [
-          {
-            "name": "origin1",
-            "properties": {
-              "hostName": "[replace(replace(reference(resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName')),'2015-06-15' ).primaryEndpoints.blob,'https://',''),'/','')]"
-            }
+      },
+      "resources": [
+        {
+          "apiVersion": "2015-06-01",
+          "name": "[variables('endpointName')]",
+          "type": "endpoints",
+          "dependsOn": [
+            "[variables('profileName')]",
+            "[variables('storageAccountName')]"
+          ],
+          "location": "[resourceGroup().location]",
+          "tags": {
+            "displayName": "[variables('endpointName')]"
+          },
+          "properties": {
+            "originHostHeader": "[replace(replace(reference(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))).primaryEndpoints.blob,'https://',''),'/','')]",
+            "isHttpAllowed": true,
+            "isHttpsAllowed": true,
+            "queryStringCachingBehavior": "IgnoreQueryString",
+            "contentTypesToCompress": [ "text/plain", "text/html", "text/css", "application/x-javascript", "text/javascript" ],
+            "isCompressionEnabled": true,
+            "origins": [
+              {
+                "name": "origin1",
+                "properties": {
+                  "hostName": "[replace(replace(reference(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))).primaryEndpoints.blob,'https://',''),'/','')]"
+                }
+              }
+            ]
           }
-        ]
-      }
+        }
+      ]
+    }
+  ],
+  "outputs": {
+    "hostName": {
+      "type": "string",
+      "value": "[reference(resourceId('Microsoft.cdn/profiles/endpoints', variables('profileName'), variables('endpointName'))).hostName]"
```

Comment:
You should be able to simplify these to: "[reference(variables('endpointName')).hostName]"